pkg/kubectl/cmd/expose.go

```diff
@@ -35,11 +36,12 @@ type ExposeOptions struct {
 }
 
 const (
-	expose_long = `Take a replication controller, service or pod and expose it as a new Kubernetes Service.
+	expose_long = `Take a replication controller, service, or pod and expose it as a new Kubernetes service.
 
-Looks up a replication controller, service or pod by name and uses the selector for that resource as the
-selector for a new Service on the specified port. If no labels are specified, the new service will
-re-use the labels from the resource it exposes.`
+Looks up a replication controller, service, or pod by name and uses the selector for that resource as the
+selector for a new service on the specified port. Note that if no port is specified via --port and the 
+exposed resource has multiple ports, all will be re-used by the new service. Also if no labels are specified,
```

Comment:
No, we can have either one port, or multiple ports, or --port. --port
stomps inheriting, but if it's unspecified and the exposed object has
multiple ports, the new service will inherit all of them.

On Wed, Oct 14, 2015 at 5:00 PM, Brian Grant <notifications@github.com>
wrote:

> In pkg/kubectl/cmd/expose.go
> <https://github.com/kubernetes/kubernetes/pull/15461#discussion_r42004816>
> :
>
> >
> > -Looks up a replication controller, service or pod by name and uses the selector for that resource as the
> > -selector for a new Service on the specified port. If no labels are specified, the new service will
> > -re-use the labels from the resource it exposes.`
> > +Looks up a replication controller, service, or pod by name and uses the selector for that resource as the
> > +selector for a new service on the specified port. Note that if no port is specified via --port and the
> > +exposed resource has multiple ports, all will be re-used by the new service. Also if no labels are specified,
>
> This makes it sound as if --port must be specified in the case that the
> resource has only one port. Is that true?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/15461/files#r42004816>.
>